homeassistant/components/smarttub/config_flow.py

```diff
@@ -24,30 +20,74 @@ class SmartTubConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     VERSION = 1
     CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
 
+    def __init__(self) -> None:
+        """Instantiate config flow."""
+        super().__init__()
+        self._reauth_input = None
+        self._reauth_entry = None
+
     async def async_step_user(self, user_input=None):
         """Handle a flow initiated by the user."""
         errors = {}
 
-        if user_input is None:
-            return self.async_show_form(
-                step_id="user", data_schema=DATA_SCHEMA, errors=errors
-            )
+        data_schema = vol.Schema(
+            {vol.Required(CONF_EMAIL): str, vol.Required(CONF_PASSWORD): str}
+        )
+
+        if user_input is not None:
+            controller = SmartTubController(self.hass)
+            try:
+                account = await controller.login(
+                    user_input[CONF_EMAIL], user_input[CONF_PASSWORD]
+                )
+
+                await self.async_set_unique_id(account.id)
 
-        controller = SmartTubController(self.hass)
-        try:
-            account = await controller.login(
-                user_input[CONF_EMAIL], user_input[CONF_PASSWORD]
+                if self._reauth_input is not None:
```

Comment:
Move the code that can't raise out of the try.  Something like

```python
            try:
                account = await controller.login(
                    user_input[CONF_EMAIL], user_input[CONF_PASSWORD]
                )
            except LoginFailed:
                errors["base"] = "invalid_auth"
            else:
                await self.async_set_unique_id(account.id)
                if self._reauth_input is None:
                    self._abort_if_unique_id_configured()
                    return self.async_create_entry(
                        title=user_input[CONF_EMAIL], data=user_input
                    )
                # this is a reauth attempt
                if self._reauth_entry.unique_id != self.unique_id:
                    # there is a config entry matching this account, but it is not the one we were trying to reauth
                    return self.async_abort(reason="already_configured")
                self.hass.config_entries.async_update_entry(
                    self._reauth_entry, data=user_input
                )
                await self.hass.config_entries.async_reload(self._reauth_entry.entry_id)
                return self.async_abort(reason="reauth_successful")
```

Reply:
Nice, I didn't know about except/else